src/main/java/org/apache/mesos/executor/CustomExecutor.java

```diff
@@ -111,6 +115,45 @@ public void launchTask(final ExecutorDriver driver, final Protos.TaskInfo task)
         }
     }
 
+    private void executeHealthCheck(Protos.TaskInfo taskInfo, ExecutorTask executorTask) {
+        if (!taskInfo.hasHealthCheck()) {
+            LOGGER.info("No health check for task: " + taskInfo.getName());
+            return;
+        }
+
+        try {
+            HealthCheckMonitor healthCheckMonitor =
+                    new HealthCheckMonitor(
+                            HealthCheckHandler.create(
+                                    taskInfo,
+                                    scheduledExecutorService,
+                                    new HealthCheckStats(taskInfo.getName())),
+                            executorTask);
+            LOGGER.info("Submitting health check monitor.");
+            Future<Optional<HealthCheckStats>> futureOptionalHealthCheckStats =
+                    executorService.submit(healthCheckMonitor);
+
+            executorService.submit(new Runnable() {
```

Comment:
could this logging wrapper be placed inside of HealthCheckMonitor? it's not strictly operating within the monitor process itself but it's pretty closely tied to its output